src/value/owned.rs

```diff
@@ -472,7 +546,30 @@ mod value_tests {
                 assert_eq!(Some("goodbye".as_bytes()), e.as_bytes());
             }
         ),
-        // TODO add cases for list/sexp/struct
+        case::list(
+            OwnedValue::List(OwnedSequence::new(vec![OwnedElement::from(OwnedValue::Boolean(true)), OwnedElement::from(OwnedValue::Boolean(false))])),
+            IonType::List,
+            AsSequence,
+            &|e: &OwnedElement| {
+                assert_eq!(Some(&OwnedSequence::new(vec![OwnedElement::from(OwnedValue::Boolean(true)), OwnedElement::from(OwnedValue::Boolean(false))])), e.as_sequence());
+            }
+        ),
+        case::sexp(
+            OwnedValue::SExpression(OwnedSequence::new(vec![OwnedElement::from(OwnedValue::Boolean(true)), OwnedElement::from(OwnedValue::Boolean(false))])),
+            IonType::SExpression,
+            AsSequence,
+            &|e: &OwnedElement| {
+                assert_eq!(Some(&OwnedSequence::new(vec![OwnedElement::from(OwnedValue::Boolean(true)), OwnedElement::from(OwnedValue::Boolean(false))])), e.as_sequence());
```

Comment:
IIRC, we have an `FromIterator` implementation that should make this a little more compact